src/genetics.rs

```diff
@@ -76,32 +74,29 @@ pub fn run_simulation<const N: usize, const M: usize>(
     base: &Board<N>,
     population: ArrayVec<Board<N>, M>,
 ) -> Result<Board<N>, NoSolutionFound<N, M>> {
-    let population_scores: ArrayVec<(Board<N>, u8), M> = ArrayVec::new_const();
-    let population_scores = Mutex::<ArrayVec<(Board<N>, u8), M>>::new(population_scores);
-    let valid_solutions = population
+    let population_scores: Result<Vec<(Board<N>, u8)>, Board<N>> = population
```

Comment:
I like this a lot better, agreed that it helps readability and avoid the `Mutex` contention! :+1: